main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -37,18 +37,49 @@ object Typer extends Phase[ResolvedAst.Root, TypedAst.Root] {
     * Type checks the given AST root.
     */
   def run(root: ResolvedAst.Root)(implicit flix: Flix): Validation[TypedAst.Root, CompilationError] = flix.phase("Typer") {
+    val classesVal = visitClasses(root)
     val defsVal = visitDefs(root)
     val enumsVal = visitEnums(root)
     val latticeOpsVal = visitLatticeOps(root)
     val propertiesVal = visitProperties(root)
 
-    Validation.mapN(defsVal, enumsVal, latticeOpsVal, propertiesVal) {
-      case (defs, enums, latticeOps, properties) =>
+    Validation.mapN(classesVal, defsVal, enumsVal, latticeOpsVal, propertiesVal) {
+      case (classes, defs, enums, latticeOps, properties) =>
         val specialOps = Map.empty[SpecialOperator, Map[Type, Symbol.DefnSym]]
-        TypedAst.Root(defs, enums, latticeOps, properties, specialOps, root.reachable, root.sources)
+        TypedAst.Root(classes, defs, enums, latticeOps, properties, specialOps, root.reachable, root.sources)
     }
   }
 
+  /**
```

Comment:
I have a fundamental question: Given that type classes do not (yet) have implementations (expressions). What is there to type check about the classes themselves?

Reply:
Looks like the only part that needs type checking is the annotations. I assume that checking is done for the space/time annotations, which wouldn't make sense in a typeclass, I think. But we could later have other annotations that would make sense on sigs.

Reply:
OK